### ISSUE TITLE
Disabling beach gesture broken on OxygenOS 12

### ISSUE THREAD

**Issue by ztNFny** [Magisk-Modules-Alt-Repo/HideNavBar#25]
On Oneplus 7 and 9 with Oxygen OS 12 disabling the back gesture doesn't work, neither left-only nor both.

Please let me know if you need any logs or apks.

**Comment by DanGLES3** [comment]
> On Oneplus 7 and 9 with Oxygen OS 12 disabling the back gesture doesn't work, neither left-only nor both.
> 
> Please let me know if youyou need any logs or apks.



> On Oneplus 7 and 9 with Oxygen OS 12 disabling the back gesture doesn't work, neither left-only nor both.
> 
> Please let me know if you need any logs or apks.

I'll see what I can do, tho I suspect this might be because OOS 12 uses a custom back gesture implementation that isn't tied to any settings

**Comment by DanGLES3** [comment]
> On Oneplus 7 and 9 with Oxygen OS 12 disabling the back gesture doesn't work, neither left-only nor both.
> 
> Please let me know if you need any logs or apks.

Could you try again on the latest version?

**Comment by M7MDE7EGAZY** [comment]
doesn't work on realme gt 2 pro completely

**Comment by M7MDE7EGAZY** [comment]
android 14

**Comment by DanGLES3** [comment]
> doesn't work on realme gt 2 pro completely

Can guess why, ColorOS likely doesn't use the stock Android value for back gesture sensitivity, I'll have to look into the code of ColorOS to see if I can find anything

**Comment by M7MDE7EGAZY** [comment]
if you need anything i can send it

**Comment by DanGLES3** [comment]
> if you need anything i can send it

Systemui and framework apks would be nice

**Comment by M7MDE7EGAZY** [comment]
don't know what is framework but i sent 2 apks + systemui one is android & the other is oplus

**Comment by M7MDE7EGAZY** [comment]
are those the ones ?

**Comment by M7MDE7EGAZY** [comment]
https://we.tl/t-g7EtJSEStO try this from latest a14 beta coloros

**Comment by M7MDE7EGAZY** [comment]
Thanks a lot

**Comment by M7MDE7EGAZY** [comment]
are those the correct ones ?

**Comment by DanGLES3** [comment]
> are those the correct ones ?

Yes, tho it's hard to find what controls the back gesture sensitivity of ColorOS/Oxygen OS in any way that I could disable with Magisk

**Comment by M7MDE7EGAZY** [comment]
i use normal sensitivity but full screen is needed a lot the space under keyboard is irritating

**Comment by M7MDE7EGAZY** [comment]
still no hope ?

**Comment by DanGLES3** [comment]
> still no hope ?

Still haven't found any way to disable the back gestures of Color OS, I'm still looking into it but mostly seem like they hardcoded the way the back gesture works

**Comment by M7MDE7EGAZY** [comment]
what about full screen and removing that shitty space under keyboard ?

**Comment by DanGLES3** [comment]
> what about full screen and removing that shitty space under keyboard ?

Is that also broken on Color OS?

**Comment by M7MDE7EGAZY** [comment]
yes just a sec i'll send ss

**Comment by M7MDE7EGAZY** [comment]
![Uploading Screenshot_2023-12-20-01-46-35-66_6012fa4d4ddec268fc5c7112cbb265e7.png…]()

**Comment by M7MDE7EGAZY** [comment]
![Screenshot_2023-12-20-01-46-35-66_6012fa4d4ddec268fc5c7112cbb265e7](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/assets/76889809/2ac932ca-c83a-44d4-8f75-3f5914027db8)

**Comment by M7MDE7EGAZY** [comment]
![image](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/assets/76889809/66b863e8-07a0-4efb-802a-2a9c551b5559)

**Comment by DanGLES3** [comment]
> ![Screenshot_2023-12-20-01-46-35-66_6012fa4d4ddec268fc5c7112cbb265e7](https://private-user-images.githubusercontent.com/76889809/291740634-2ac932ca-c83a-44d4-8f75-3f5914027db8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTEiLCJleHAiOjE3MDMwMzAwMjUsIm5iZiI6MTcwMzAyOTcyNSwicGF0aCI6Ii83Njg4OTgwOS8yOTE3NDA2MzQtMmFjOTMyY2EtYzgzYS00NGQ0LThmNzUtM2Y1OTE0MDI3ZGI4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFJV05KWUFYNENTVkVINTNBJTJGMjAyMzEyMTklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjMxMjE5VDIzNDg0NVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWYwOWQzMGI4ZGMwNTg3ZDJiMTE0Nzg1ODU0YjNlM2UwN2EyNjUzZGMyOWU4MjJkMzMxZTNiZDQwMzhkMzg0YTQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.0hFf8u2I4YjVyFUK6IXsjGuSev0qnYKy4ZSSbR-D2BU)

This space seems to come from Gboard's own spacing feature, have you tried using something like Rboard to reduce the bottom padding of the keyboard?

**Comment by M7MDE7EGAZY** [comment]
i used this module in cos 13.1 it was removed after choosing full screen but now like that

**Comment by M7MDE7EGAZY** [comment]
I'll try rbosrd

**Comment by M7MDE7EGAZY** [comment]
yes exactly

**Comment by DanGLES3** [comment]
> yes exactly

Try the following command on a terminal app and see if it does anything, found it looking at the decompiled systemui code

su -c settings put secure hide_navigationbar_enable 1

**Comment by DanGLES3** [comment]
> i never use gcam nor disable back gesture

Also, does Color OS have its own immersive/hide gesture bar option? that might conflict with my module

**Comment by M7MDE7EGAZY** [comment]
![Screenshot_2023-12-20-11-39-09-82_fc704e6b13c4fb26bf5e411f75da84f2](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/assets/76889809/7fd65d2f-a752-4ee2-bf11-6720b956b844)
![Screenshot_2023-12-20-11-39-02-04_fc704e6b13c4fb26bf5e411f75da84f2](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/assets/76889809/cd4aa5ee-7cda-4af8-a288-7581d1baf5d3)
![Screenshot_2023-12-20-11-38-54-54_fc704e6b13c4fb26bf5e411f75da84f2](https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/assets/76889809/e1507fca-8d87-4185-8a19-a49d6d80093a)

**Comment by M7MDE7EGAZY** [comment]
now after small update the command worked 
su -c settings put secure hide_navigationbar_enable 1

**Comment by M7MDE7EGAZY** [comment]
but now back gesture broken & in navigation it changed to buttons if i enabled gesture the space returns

**Comment by M7MDE7EGAZY** [comment]
https://github.com/Magisk-Modules-Alt-Repo/HideNavBar/assets/76889809/a7b89ca0-ca1f-4583-8cf4-7f925efc2d44

**Comment by M7MDE7EGAZY** [comment]
Immersion can be done, but I don't know why it doesn't work if you remove the bottom. The solution of removing the bottom works on Android14QPR, but it doesn't work on Color os14. My guess is that color replaces the bottom with dimen.
